tests/common/plugins/memory_utilization/README.md

```diff
@@ -0,0 +1,573 @@
+  - The dependency file also could includes a "HWSKU" key for all hwsku related memory items. If the hwsku memory item is same as a common one, it will overwrite it.
+  - This dependency file should be left empty in the public branch, with special configurations added in internal branch.
+
+Memory utilization config files include memory items which need to check.
+Each memory item include "name", "cmd", "memory_params" and "memory_check".
+- **name**:          The name of the memory check item.
+- **cmd**:           The shell command is run on the DUT to collect memory information.
+- **memory_params**: The items and thresholds for memory usage, defined as a Dict type in the configuration JSON file. It could be modified in the test case.
+- **memory_check**:  The function used to parse the output of the shell command takes two input parameters: cmd's output string and memory_params. It returns the parsered memory inforamtion, which will be compared with "memory_params" to check for memory threshold.
+
+#### Workflow
+1. Collect memory information based on memory utilization config files before running a test case.
+    - Execute the "cmd" and use the "memory_check" function to collect the memory information before running the test case.
+2. Collect memory information based on memory utilization config files after the test case is completed.
+    - Execute the "cmd" and use the "memory_check" function to collect the memory information after running the test case.
+3. Compare the memory information from before and after the test run.
+    - Compare the collected memory information with the thresholds in "memory_params".
+4. Raise an alarm if there is any memory leak or if the memory usage exceeds the high memory threshold based on the memory utilization config files.
+    ```
+    >       pytest.fail(message)
+    E       Failed: [ALARM]: monit:memory_usage memory usage 77.8 exceeds high threshold 70.0
+    ```
+
+
+### Memory Utilization usage example
+
+Below is a description of the possible uses for the "memory_utilization" fixture/module.
+
+##### memory_utilization fixture
+In the root conftest there is an implemented "memory_utilization" pytest fixture that starts automatically for all test cases.
+The main flow of the fixture is as follows:
+- memory_utilization collects memory information before the test case starts.
+- memory_utilization collects memory information after the test case finishes.
+- memory_utilization compares DUT memory usage and displays the results.
+- if memory_utilization finds any exceeded thresholds for high memory usage or memory increase, it will display the result and pytest will generate an 'error'.
+
+#### To skip memory_utilization for:
+- all test cases - use pytest command line option ```--memory_utilization```
+- specific test case: mark test case with ```@pytest.mark.memory_utilization``` decorator. Example is shown below.
```

Comment:
@lipxu should be disable_memory_utilization decorator?

Reply:
thanks for reminder, fixed.